### ISSUE TITLE
Don't rely type_of_target, it is pretty buggy

### ISSUE THREAD

**Issue by mitar** [automl/auto-sklearn#293]
```python
import sklearn.datasets
from sklearn.metrics import classification as classification_metrics
dataset = sklearn.datasets.load_linnerud()
classification_metrics.type_of_target(dataset.target)
```

Returns `multiclass-multioutput` instead `continuous-multioutput`.

Similarly the following return `multiclass`, but it should `regression`.

```python
import sklearn.datasets
from sklearn.metrics import classification as classification_metrics
dataset = sklearn.datasets.load_diabetes()
classification_metrics.type_of_target(dataset.target)
```

I think `AutoSklearnClassifier` and `AutoSklearnRegressor` should just be replaced with one, `AutoModel` or something which simply tries to run both classifier and regressor and see what works better.

**Comment by mitar** [comment]
Trying to run regression on `linnerud` dataset also does not work because of this function:

```
[WARNING] [2017-05-29 20:17:10,973:EnsembleBuilder(1):95574e901f03672766c05ee1baef1335] Error loading predictions_ensemble_1_00001.npy: <class 'ValueError'> - multiclass-multioutput
[ERROR] [2017-05-29 20:17:10,975:EnsembleBuilder(1):95574e901f03672766c05ee1baef1335] Caught ValueError: multiclass-multioutput
```

I run it as a regression, but because internally it detects it as `multiclass-multioutput`, it fails.

**Comment by mfeurer** [comment]
I'll have a look at this issue next week (sorry, I don't have time to check this before). Maybe auto-sklearn should not use the `type_of_target` function for regression?

**Comment by amueller** [comment]
@mitar which version of scikit-learn is that? Maybe open an issue there?

**Comment by mfeurer** [comment]
The underlying issue appeares to be solved by #506 - please reopen if this is still an issue.
